src/contextify.cc

```diff
@@ -180,170 +182,201 @@ class ContextWrap : public ObjectWrap {
         return true;
     }
 
-    static NAN_PROPERTY_GETTER(GlobalPropertyGetter) {
-        NanScope();
-        Local<Object> data = args.Data()->ToObject();
+    static void GlobalPropertyGetter(Local<String> property, const PropertyCallbackInfo<Value>& info) {
+        Local<Object> data = info.Data()->ToObject();
         ContextifyContext* ctx = ObjectWrap::Unwrap<ContextWrap>(data)->ctx;
-        if (!ctx)
-            NanReturnUndefined();
-        Local<Value> rv = NanNew(ctx->sandbox)->GetRealNamedProperty(property);
-        if (rv.IsEmpty()) {
-            rv = NanNew(ctx->proxyGlobal)->GetRealNamedProperty(property);
+
+        if (!ctx) {
+            info.GetReturnValue().SetUndefined();
+            return;
         }
-        NanReturnValue(rv);
+
+        Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
+
+//        if (rv.IsEmpty()) {
+//            rv = Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property);
+//        }
+
+        info.GetReturnValue().Set(rv);
     }
 
-    static NAN_PROPERTY_SETTER(GlobalPropertySetter) {
-        NanScope();
-        Local<Object> data = args.Data()->ToObject();
+    static void GlobalPropertySetter(Local<String> property, Local<Value> value, const PropertyCallbackInfo<Value>& info) {
```

Comment:
I was thinking v8::Context::Scope could be replaced with Nan::ContextScope.

For versions which have v8::Context::Scope, Nan::ContextScope would just be a 
wrapper class composed of a v8::Context::Scope. The constructor of 
Nan::ContextScope would initialize the v8::Context::Scope member, the default 
destructor would destroy the v8::Context::Scope, thus exiting the scope.

For older versions, the constructor of Nan::ContextScope would call 
v8::Context::Enter and the destructor would call v8::Context::Exit. Here, the 
v8::Context supplied in the constructor would need to be stored as a 
v8::Local<v8::Context> member.

See Nan::HandleScope for reference
https://github.com/nodejs/nan/blob/master/nan.h#L336-L359

Although, now I see that this is exactly what v8::Context::Scope does:
https://github.com/nodejs/node/blob/master/deps/v8/include/v8.h#L6716-L6725

So the old v8::Context::Enter / v8::Context::Exit syntax should still work 
fully, and it seems v8::Context::Scope has been around forever 
https://github.com/nodejs/node-v0.x-archive/blob/v0.8.28/deps/v8/include/v8.h#L3624-L3632 so I don't think 
anything actually needs to be done. Everything should work already.

On Tuesday 03 November 2015 02:12:06 Ido Ben-Yair wrote:
> >      }
> > 
> > -    static NAN_PROPERTY_SETTER(GlobalPropertySetter) {
> > -        NanScope();
> > -        Local<Object> data = args.Data()->ToObject();
> > +    static void GlobalPropertySetter(Local<String> property, Local<Value>
> > value, const PropertyCallbackInfo<Value>& info) {
> Only that it didn't seem to work and I didn't want to spend too much time
> trying to figure out why, I needed to move fast. But I can try again now
> that I got Contextify to work with node 5.0.0.
> 
> I'd prefer to add the new functionality to NAN and avoid having ifdefs in a
> client project like Contextify, if you're willing to elaborate.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/brianmcd/contextify/pull/190/files#r43731733